crates/ludtwig-parser/src/syntax/untyped.rs

```diff
@@ -446,6 +452,11 @@ pub enum SyntaxKind {
     TWIG_CACHE_STARTING_BLOCK,
     TWIG_CACHE_ENDING_BLOCK,
 
+    // Drupal Trans
+    TWIG_TRANS,
+    TWIG_TRANS_BLOCK,
+    TWIG_ENDTRANS_BLOCK,
```

Comment:
As mentioned above, for the syntax nodes I would suggest something similar to the basic twig block:
```rust

    // twig block
    TWIG_BLOCK,
    TWIG_STARTING_BLOCK,
    TWIG_ENDING_BLOCK,
    
     // twig sandbox
    TWIG_SANDBOX,
    TWIG_SANDBOX_STARTING_BLOCK,
    TWIG_SANDBOX_ENDING_BLOCK,
```
So after seeing how the twig code for this looks like I would propose:
```rust
     // Drupal Trans
    TWIG_TRANS,
    TWIG_TRANS_STARTING_BLOCK,
    TWIG_TRANS_ENDING_BLOCK,
```

Reply:
One other thing I forgot to mention, you should add AST structs for these. Have a look here https://github.com/MalteJanz/ludtwig/blob/da850a17d0d3b72f0bca94a94b33f25a052e5a36/crates/ludtwig-parser/src/syntax/typed.rs#L523-L531

You can also add utility methods if you like (can be useful to work with that AST node in rules that check specific things about these `trans` blocks), see for example the AST implementation for twig blocks:

https://github.com/MalteJanz/ludtwig/blob/da850a17d0d3b72f0bca94a94b33f25a052e5a36/crates/ludtwig-parser/src/syntax/typed.rs#L82-L137